dace/transformation/auto/auto_optimize.py

```diff
@@ -166,17 +166,28 @@ def tile_wcrs(graph_or_subgraph: GraphViewType,
         # Transform all outgoing WCR and stream edges
         mapexit = graph.exit_node(mapentry)
         outer_mapexit = graph.exit_node(outer_mapentry)
+
+        # Tuple of (transformation type, options, pattern)
+        to_apply: Tuple[Union[dataflow.StreamTransient,
+                              dataflow.AccumulateTransient], Dict[str, Any],
+                        Dict[str, nodes.Node]] = None
         for e in graph.out_edges(mapexit):
             if isinstance(sdfg.arrays[e.data.data], dt.Stream):
                 mpath = graph.memlet_path(e)
                 tasklet = mpath[0].src
                 if not isinstance(tasklet, nodes.Tasklet) or len(mpath) != 3:
                     # TODO(later): Implement StreamTransient independently of tasklet
                     continue
-                dataflow.StreamTransient.apply_to(sdfg,
-                                                  tasklet=tasklet,
-                                                  map_exit=mapexit,
-                                                  outer_map_exit=outer_mapexit)
+
+                # Make transient only if there is one WCR/stream
+                if to_apply is not None:
+                    to_apply = None
+                    break
+
+                to_apply = (dataflow.StreamTransient, {},
+                            dict(tasklet=tasklet,
+                                 map_exit=mapexit,
+                                 outer_map_exit=outer_mapexit))
```

Comment:
This snuck in? ;-)

Reply:
1. it's part of the PR description
2. it appeared on auto-optimized `@dace.program`s